### PR TITLE
Add NamespaceMap CRD

### PR DESCRIPTION
**Pull request by Andreagit97**
# Description

This PR add a new CRD `NamespaceMap`, which will replace `NamespaceNattingTable` resource.
Main differences between `NamespaceNattingTable` and `NamespaceMap`:

- `NamespaceMap` is a Namespaced resource.
- `NamespaceMap` has only `NattingTable`and no `DeNattingTable` (moved from `Spec` to `Status`).
- `NamespaceMap` has no more `RemoteClusterId` in its `Spec` (no more useful, we will use a specific label to get it).

# Fixes
**STEP 1** #572

### CODE REVIEW

**Review by palexster — APPROVED** [review]

File: apis/virtualKubelet/v1/groupversion_info.go
Comment (giorio94) [inline apis/virtualKubelet/v1/groupversion_info.go:20]: I suppose this should be v1alpha1

File: apis/virtualKubelet/v1/groupversion_info.go
Comment (giorio94) [inline apis/virtualKubelet/v1/groupversion_info.go:29]: Same here

Comment (giorio94) [inline apis/virtualKubelet/v1/groupversion_info.go:29]: Is the folder where these files reside correct? I suppose not since the group is not virtualkubelet.liqo.io

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (giorio94) [inline apis/virtualKubelet/v1/namespacemap_types.go:30]: Is it correct the spec is empty? I feel it pretty strange

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (giorio94) [inline apis/virtualKubelet/v1/namespacemap_types.go:37]: Each non-standard field should be preceded by a comment to document it, both in the code and in the generated CRD.

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (giorio94) [inline apis/virtualKubelet/v1/namespacemap_types.go:36]: This should be deleted, or an empty line left below, otherwise it appears in the generated CRD

**Review by giorio94 — COMMENTED** [review]
Some comments inline

File: apis/virtualKubelet/v1/groupversion_info.go
Comment (Andreagit97) [reply apis/virtualKubelet/v1/groupversion_info.go:20]: yes, i have to change it, thank you

File: apis/virtualKubelet/v1/groupversion_info.go
Comment (Andreagit97) [reply apis/virtualKubelet/v1/groupversion_info.go:29]: Thanks

Comment (Andreagit97) [reply apis/virtualKubelet/v1/groupversion_info.go:29]: yes, i have to change it, thank you

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (Andreagit97) [reply apis/virtualKubelet/v1/namespacemap_types.go:30]: At the moment we have nothing to implement here, so I left it empty... Should I remove it ?

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (Andreagit97) [reply apis/virtualKubelet/v1/namespacemap_types.go:37]: My fault, thank you very much

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (Andreagit97) [reply apis/virtualKubelet/v1/namespacemap_types.go:36]: I didn't know it, thank you very much

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (giorio94) [reply apis/virtualKubelet/v1/namespacemap_types.go:30]: I don't think you can delete the spec entirely. I was just raising a concern that I felt strange to have a CRD with no spec and a status, but if it comes from a previous design phase and the @liqotech/maintainers agreed that is fine to me

Comment (Andreagit97) [reply apis/virtualKubelet/v1/namespacemap_types.go:30]: Thank you very much for the advise, we will consider again it.

File: deployments/liqo/crds/namespaceresources.liqo.io_namespacemaps.yaml
Comment (giorio94) [inline deployments/liqo/crds/namespaceresources.liqo.io_namespacemaps.yaml:1]: This CRD should be deleted

**Review by giorio94 — COMMENTED** [review]
Please, squash the commits before merging. Additionally, do not add `fixes #issue` in the description of the PR, otherwise it will be closed (I do not think this is your goal)

File: apis/virtualKubelet/v1/namespacemap_types.go
Comment (mlavacca) [reply apis/virtualKubelet/v1/namespacemap_types.go]: I agree it is pretty strange. For the time being, we don't have any spec in this API, but most likely we'll add some fields in the future. Since this API represents the current mapping between local and remote namespaces, we used only the status subresource.

File: deployments/liqo/crds/namespaceresources.liqo.io_namespacemaps.yaml
Comment (Andreagit97) [reply deployments/liqo/crds/namespaceresources.liqo.io_namespacemaps.yaml]: Sorry, thank you very much @giorio94

**Review by mlavacca — APPROVED** [review]

**Pull request merged**
